release-notes/swan-lake-alpha3-release-note.md

```diff
@@ -144,6 +144,451 @@ io:println(nameText); // "DanBrown"
+}
+```
+
+##### Security Updates
+
+Renamed the `ballerina/encoding` module as `ballerina/url` and updated the APIs.
+```ballerina
+import ballerina/url;
+
+string|url:Error encoded = url:encode("http://localhost:9090", "UTF-8");
+string|url:Error decoded = url:decode("http%3A%2F%2Flocalhost%3A9090", "UTF-8");
+```
+
+The Ballerina HTTP listener can be configured to authenticate and authorize the inbound requests with Basic Auth file user store.
+
+Improved client and listener `SecureSocket` APIs of HTTP, GRPC, WebSocket, GraphQL, WebSub, WebSubHub, TCP, Email, NATS, STAN and RabbitMQ modules.
+```ballerina
+public type ListenerSecureSocket record {|
+   crypto:KeyStore|CertKey key;
+   record {|
+       VerifyClient verifyClient = REQUIRE;
+       crypto:TrustStore|string cert;
+   |} mutualSsl?;
+   record {|
+       Protocol name;
+       string[] versions = [];
+   |} protocol?;
+   record {|
+       CertValidationType type = OCSP_STAPLING;
+       int cacheSize;
+       decimal cacheValidityPeriod;
+   |} certValidation?;
+   string[] ciphers = [];
+   boolean shareSession = true;
+   decimal handshakeTimeout?;
+   decimal sessionTimeout?;
+|};
+
+public type ClientSecureSocket record {|
+   boolean enable = true;
+   crypto:TrustStore|string cert?;
+   crypto:KeyStore|CertKey key?;
+   record {|
+       Protocol name;
+       string[] versions = [];
+   |} protocol?;
+   record {|
+       CertValidationType type = OCSP_STAPLING;
+       int cacheSize;
+       decimal cacheValidityPeriod;
+   |} certValidation?;
+   string[] ciphers?;
+   boolean verifyHostName = true;
+   boolean shareSession = true;
+   decimal handshakeTimeout?;
+   decimal sessionTimeout?;
+|};
+
+public type CertKey record {|
+   string certFile;
+   string keyFile;
+   string keyPassword?;
+|};
+ 
+public enum VerifyClient {
+   REQUIRE,
+   OPTIONAL
+}
+ 
+public enum Protocol {
+   SSL,
+   TLS,
+   DTLS
+}
+ 
+public enum CertValidationType {
+   OCSP_CRL,
+   OCSP_STAPLING
+}
+```
+
+Improved `SecureSocket` configuration of JDK11 client of JWT and OAuth2 modules.
+
+Added support for OAuth2 client authentication of JDK11 client, which is used to call authorization endpoint.
+
+##### TCP Package Updates
+
+Introduce SSL/TLS support for both the client and listener.
+```ballerina
+import ballerina/tcp;
+
+configurable string certPath = ?;
+
+public function main() returns error? {
+    tcp:Client socketClient = check new ("localhost", 9002, secureSocket = {
+        cert: certPath,
+        protocol: {
+            name: tcp:TLS,
+            versions: ["TLSv1.2", "TLSv1.1"]
+        },
+        ciphers: ["TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA"]
+    });
+
+    string msg = "Hello Ballerina Echo from secure client";
+    byte[] msgByteArray = msg.toBytes();
+    check socketClient->writeBytes(msgByteArray);
+
+    readonly & byte[] receivedData = check socketClient->readBytes();
+    test:assertEquals('string:fromBytes(receivedData), msg, "Found unexpected output");
```

Comment:
We don't need this line, right? We aren't (and shouldn't IMO) importing ballerina/test here.